Market Tools/macd_strategy.py

```diff
@@ -0,0 +1,147 @@
+# Import the required libraries
+import numpy as np
+import pandas as pd
+import hvplot.pandas
+from pathlib import Path
+from datetime import date
+from pandas.core.frame import DataFrame
+import yfinance as yf
+import os
+import pricing
+
+def analyze_macd(signals_df:DataFrame):
```

Comment:
[Low priority for this PR] : In the future consider adding function documentation in the following format:

```
def analyze macd(signals_df:DataFrame):
"""<short description>
<one blank line>
<long description>
<one blank line>
Parameters
-------------
<parameter name>: <data type>,<required or not required>
     <description of parameter>
<parameter name2>
etc.
<one blank line>
Returns
----------
<return value 1>: <description>
"""
```